### DISCUSSION TITLE
GitHub gists

### DISCUSSION THREAD

**Discussion by patriksvensson** [cake-build/cake#3537] [Ideas]
Would be useful to be able to reference a gist from the cake script.
The source of the gist would be inserted into the build script. Good for sharing utility methods. 

``` csharp
#gist "https://gist.github.com/patriksvensson/872bfcc44b234b12ea8c"
```

**Comment by RichiCoder1** [comment]
Would the idea here be to enable streaming a file into the build script, or specifically allow the loading of full on gists?

**Comment by patriksvensson** [comment]
I'm thinking that it should work the same way as the `#l`-directive, but instead of streaming from disc, stream over the interwebz.

**Comment by devlead** [comment]
Gists after just GIT repositories so perhaps do an generic GIT repo script include?

Something like
`#git https://gist.github.com/872bfcc44b234b12ea8c.git?script=RelativeGlobberPathToFileOrFiles`
Then clone into folder relative to cake (like addins)

Then it could be cached if many scripts were to use the same repo, and just pull if exist to get latest version.

Built in GIT support for clone, pull,  push, add & commit would be really useful in general,  especially as many projects are published via GIT.

**Comment by patriksvensson** [comment]
Hmmm, maybe. Problem is that we probably would need an additional dependency on `libgit2sharp` since it's a pain to locate the git-executable. When I think about it, maybe it's better to do this as an addin when the `#nuget`-support is in place?

**Comment by devlead** [comment]
Meanwhile could just add HTTP support to the `#l` directive, both GITHub repos and gists support raw downloading if files.

**Comment by patriksvensson** [comment]
@devlead I actually like this idea of extending the `#l` directive to support downloading via HTTP.

**Comment by devlead** [comment]
@Redth this basically is what you've started to look at right?

**Comment by Redth** [comment]
Yes. This is what I started doing...
